lib/ace/ext/papeeria/tex_completer.coffee

```diff
@@ -55,66 +78,31 @@ define((require, exports, module) ->
             """
       meta: "list"
     }
-  equationSnippets = for env in equationEnvironments
-    {
-      caption: "\\begin{#{env}}..."
-      snippet: """
-                  \\begin{#{env}}
-                  \t$1
-                  \\end{#{env}}
-              """
-      meta: "equation"
-    }
-  formulasSnippets = [
-    {
-      caption: "\\frac{num}{denom}"
-      snippet: """
-                \\frac{${1:num}}{${2:denom}}
-            """
-      meta: "equation"
-    }
-    {
-      caption: "\\sum{n}{i=..}{..}"
-      snippet: """
-                  \\sum^{${1:n}}_{${2:i=1}}{${3}}
-            """
-      meta: "equation"
-    }
-  ]
-
-  equationKeywords = ["\\alpha"]
+  listSnippets = listSnippets.concat([referenceSnippet])
   listKeywords = ["\\item"]
-
   listKeywords = listKeywords.map((word) ->
     caption: word,
     value: word
     meta: "list"
   )
-  equationKeywords = equationKeywords.map((word) ->
-    caption: word,
-    value: word
-    meta: "equation"
-  )
-
-  # Specific for token"s system of type in ace
-  # We saw such a realization in html_completions.js
-  isType = (token, type) ->
-    return token.type.split(".").indexOf(type) > -1
-
 
   init = (editor, bindKey) ->
     keyboardHandler = new HashHandler.HashHandler()
     keyboardHandler.addCommand(
       name: "add item in list mode"
       bindKey: bindKey
       exec: (editor) ->
-        pos = editor.getCursorPosition()
-        curLine = editor.session.getLine(pos.row)
-        indentCount = LatexParsingContext.getNestedListDepth(editor.session, pos.row)
-        tabString = editor.getSession().getTabString()
-        # it"s temporary fix bug with added \item before \begin{itemize|enumerate}
-        if LatexParsingContext.getContext(editor.session, pos.row) == LIST_STATE && curLine.indexOf("begin") < pos.column
-          editor.insert("\n" + tabString.repeat(indentCount) + "\\item ")
+        cursor = editor.getCursorPosition();
+        line = editor.session.getLine(cursor.row);
+        tabString = editor.session.getTabString();
+        indentString = line.match(/^\s*/)[0];
+        indexOfBegin = line.indexOf("begin")
+
+        if LatexParsingContext.getContext(editor.session, cursor.row) == LIST_STATE &&  indexOfBegin < cursor.column
```

Comment:
any sense to make `getContext` returning a range which starts from \begin start position? This will eliminate the need in this hackety hack with `indexOf "begin"`